vsintegration/src/FSharp.Editor/Common/CommonHelpers.fs

```diff
@@ -882,4 +718,951 @@ module internal SymbolUse =
+                        else acc) []
+            | ParsedInput.SigFile(ParsedSigFileInput(_, _, _, _, modules)) ->
+                let rec walkModuleOrNamespaceSig idents (decls, moduleRange) =
+                    decls
+                    |> List.fold (fun acc ->
+                        function
+                        | SynModuleSigDecl.NestedModule (componentInfo, _, nestedModuleDecls, nestedModuleRange) ->
+                            if rangeContainsPos moduleRange pos then
+                                let (ComponentInfo(_,_,_,longIdent,_,_,_,_)) = componentInfo
+                                walkModuleOrNamespaceSig (longIdent::acc) (nestedModuleDecls, nestedModuleRange)
+                            else acc
+                        | _ -> acc) idents
+
+                modules
+                |> List.fold (fun acc (SynModuleOrNamespaceSig(longIdent, _, _, decls, _, _, _, moduleRange)) ->
+                        if rangeContainsPos moduleRange pos then
+                            walkModuleOrNamespaceSig (longIdent::acc) (decls, moduleRange) @ acc
+                        else acc) []
+        idents
+        |> List.rev
+        |> Seq.concat
+        |> Seq.map (fun ident -> ident.idText)
+        |> String.concat "."
+
+
+[<NoComparison>]
+type internal SymbolUse =
+    { SymbolUse: FSharpSymbolUse 
+      IsUsed: bool
+      FullNames: Idents[] }
+
+
+[<AutoOpen>]
+module internal TypedAstUtils =
+    open System.Text.RegularExpressions
+
+    let isAttribute<'T> (attribute: FSharpAttribute) =
+        // CompiledName throws exception on DataContractAttribute generated by SQLProvider
+        match Option.attempt (fun _ -> attribute.AttributeType.CompiledName) with
+        | Some name when name = typeof<'T>.Name -> true
+        | _ -> false
+
+    let tryGetAttribute<'T> (attributes: seq<FSharpAttribute>) =
+        attributes |> Seq.tryFind isAttribute<'T>
+
+    let hasModuleSuffixAttribute (entity: FSharpEntity) = 
+         entity.Attributes
+         |> tryGetAttribute<CompilationRepresentationAttribute>
+         |> Option.bind (fun a -> 
+              Option.attempt (fun _ -> a.ConstructorArguments)
+              |> Option.bind (fun args -> args |> Seq.tryPick (fun (_, arg) ->
+                   let res =
+                       match arg with
+                       | :? int32 as arg when arg = int CompilationRepresentationFlags.ModuleSuffix -> 
+                           Some() 
+                       | :? CompilationRepresentationFlags as arg when arg = CompilationRepresentationFlags.ModuleSuffix -> 
+                           Some() 
+                       | _ -> 
+                           None
+                   res)))
+         |> Option.isSome
+
+    let isOperator (name: string) =
+        name.StartsWith "( " && name.EndsWith " )" && name.Length > 4
+            && name.Substring (2, name.Length - 4) 
+               |> String.forall (fun c -> c <> ' ' && not (Char.IsLetter c))
```

Comment:
suggestions:

* maybe this whole expression can be better laid out?
* also could you have comment on stuff dealing with substring as those are really annoying to reverse engineer / run in the debugger in my head.

proposal:

```fsharp
   let isOperator (name: string) =
        let inline notSpaceOrLetter c = c <> ' ' && not (Char.IsLetter c)
        name.StartsWith "( "
        && name.EndsWith " )"
        && name.Length > 4
        // check something something
        && (name.Substring (2, name.Length - 4) |> String.forall notSpaceOrLetter) 
```